### DISCUSSION TITLE
Vaultwarden mariadb connection with enabled SSL on the DB

### DISCUSSION THREAD

**Discussion by Atanas-Hristov** [dani-garcia/vaultwarden#2740] [Q&A] [answered]
Hello,

I have struggled to connect the vaultwarden container to another that runs MariaDB while SSL is turned on.
Zero issues if I run the DB container without enabling SSL, however the second I enable SSL I see the following errors from the vaultwarden container:

> [WARN] Can't connect to database, retrying: DieselCon.
> [CAUSE] BadConnection(
>     "Access denied for user 'user'@'ip' (using password: YES)",
> )
> 
and the same error on the DB container logs - nothing mentions SSL however.

Any ideas of what needs to be done to enable SSL for the vaultwarden container?

PS:
      I have checked the user/password and remote connections - those are enabled already, however the app does not want to connect to the database.
      
Thank you for your feedback.

**Comment by BlackDex** [comment, marked as answer]
The current version from Diesel (the database library) we use doesn't have ssl mode support for MySQL/MariaDB. There is a PR open to support the latest v2 version which does have support for this.
Until that time probably better to use non-ssl.

**Reply by Atanas-Hristov** [reply]
I was beginning to think that way, but you have now confirmed it for me.
 
 Thank you for the feedback.

**Comment by jjmschofield** [comment]
For anyone coming across this today, [time has moved on](https://github.com/dani-garcia/vaultwarden/discussions/5758#discussion-8180919).
